art/attacks/evasion/adversarial_texture/adversarial_texture_pytorch.py

```diff
@@ -240,64 +255,134 @@ def _get_patch_mask(self, nb_samples: int) -> "torch.Tensor":
         return image_mask
 
     def _apply_texture(
-        self, videos: "torch.Tensor", patch: "torch.Tensor", foreground: Optional["torch.Tensor"]
+        self,
+        videos: "torch.Tensor",
+        patch: "torch.Tensor",
+        foreground: Optional["torch.Tensor"],
+        patch_points: Optional[np.ndarray],
     ) -> "torch.Tensor":
         """
         Apply texture over background and overlay foreground.
 
         :param videos: Video samples.
         :param patch: Patch to apply.
         :param foreground: Foreground mask.
+        :param patch_points: Array of shape (nb_frames, 4, 2) containing four pairs of integers (height, width)
+                             corresponding to the coordinates of the four corners top-left, top-right, bottom-right,
+                             bottom-left of the transformed image in the coordinate-system of the original image.
         :return: Patched videos.
         """
         import torch  # lgtm [py/repeated-import]
         import torchvision
 
         nb_samples = videos.shape[0]
+        nb_frames = videos.shape[1]
+        frame_height = videos.shape[2]
+        frame_width = videos.shape[3]
 
         image_mask = self._get_patch_mask(nb_samples=nb_samples)
         image_mask = image_mask.float()
```

Comment:
I noticed you had to type case image_mast to float at a bunch of places. Would it be easier to do it as part of _get_patch_mask?